res/raw/webxdc_wrapper.html

```diff
@@ -0,0 +1,145 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <style>
+      html,
+      body {
+        margin: 0;
+        padding: 0;
+      }
+      .iframe-container {
+        overflow: hidden;
+        height: 100vh;
+        width: 100vw;
+        position: relative;
+      }
+      .iframe-container iframe {
+        border: 0;
+        height: 100%;
+        left: 0;
+        position: absolute;
+        top: 0;
+        width: 100%;
+      }
+      #progress {
+        width: 100%;
+      }
+    </style>
+  </head>
+  <body>
+    <div id="loading">
+      <progress id="progress" max="100" value="0"></progress>
+    </div>
+    <div class="iframe-container">
+      <iframe id="frame"></iframe>
+    </div>
+    <iframe
+      id="test-isolated-sandbox-context"
+      sandbox="allow-scripts"
+      src="./sandboxed_iframe_rtcpeerconnection_check_5965668501706.html"
+      style="display: none"
+    ></iframe>
+    <script>
+      const thisUnchangable = (async () => {
+        const connections = [];
+        const loadingProgress = document.getElementById("progress");
+        const loadingDiv = document.getElementById("loading");
+        const iframe = document.getElementById("frame");
+        const isolatedContextTest = document.getElementById(
+          "test-isolated-sandbox-context"
+        );
+
+        const cert = {
+          certificates: [
+            await RTCPeerConnection.generateCertificate({
+              name: "ECDSA",
+              namedCurve: "P-256",
+            }),
+          ],
+        };
+
+        console.log("WEBRTC-WG: allocating");
+        loadingProgress.value = 0;
+        while (connections.length < 500) {
+          try {
+            connections.push(new RTCPeerConnection(cert));
+            if (connections.length%50 == 0) {
+              loadingProgress.value++;
+              await new Promise((res) => setTimeout(res));
+            }
+          } catch (error) {
+            loadingProgress.value++;
+            new Array(1024*1024).fill(0);
+            await new Promise((res) => setTimeout(res, 500));
+            console.log("WEBRTC-WG: waiting for gc");
+          }
+        }
+        console.log("WEBRTC-WG: done");
+
+        try {
+          connections.push(new RTCPeerConnection());
+          console.log("Error: was able to create more than 500 connections");
+          loadingDiv.innerText =
+            "Error: was not able to block webrtc ERROR_A";
+        } catch (error) {
```

Comment:
Nitpick: doing just `new RTCPeerConnection()` inside the `try {` block would be more correct IMO.

Reply:
i also think, it is good enough, for an attack one has to make `console.log` or `innerText` fail - before any code from the webxdc is executed.